tools/python/category/omim_parsers.py

```diff
@@ -0,0 +1,98 @@
+import re
+from typing import Optional, Tuple, Dict, List
+
+LANGUAGES = [
+    "af", "ar", "be", "bg", "ca", "cs", "da", "de", "el", "en", "en-GB", "es", "es-MX", "et",
+    "eu", "fa", "fi", "fr", "fr-CA", "he", "hi", "hu", "id", "it", "ja", "ko", "lt", "mr", "nb",
+    "nl", "pl", "pt", "pt-BR", "ro", "ru", "sk", "sv", "sw", "th", "tr", "uk", "vi", "zh-Hans", "zh-Hant"
+]
+
+
+class AbstractParser:
+    def __init__(self, keys: List[str]):
+        self.keys = keys
+
+    def parse_line(self, line: str) -> Optional[Tuple[str, str]]:
+        raise NotImplementedError("You must implement parse_line.")
+
+    def match_category(self, line: str, result: Dict[str, Dict]):
+        category_match = self.category().search(line)
+        if category_match:
+            category = category_match.group(1)
+            if category in self.keys:
+                if category not in result:
+                    result[category] = {}
+
+    def parse_file(self, filename: str) -> Dict[str, Dict]:
+        result = {}
+        current_category = None
+
+        with open(filename, "r", encoding="utf-8") as file:
+            for line in file:
+                line = line.strip()
+
+                # Skip comments and empty lines
+                if self.should_exclude_line(line):
+                    continue
+
+                # Match a new category
+                category_match = self.category().match(line)
```

Comment:
Small optimization here: no need to call `self.category()` for every line.

Please create new variable at the start of `parse_file(...)` method and put result of `self.category()`
For example:

```python
category_regex = self.category()
with open(...):
    ...
    category_match = category_regex.match(line)
```